packages/image_picker/lib/image_picker.dart

```diff
@@ -56,15 +56,24 @@ class ImagePicker {
     return path == null ? null : new File(path);
   }
 
+  /// Returns a [File] object pointing to the video that was picked.
+  ///
+  /// The [source] argument controls where the video comes from. This can
+  /// be either [ImageSource.camera] or [ImageSource.gallery].
+  ///
+  /// [maxDuration] is the maximum amount of Seconds (eg. 30.0) that is available to be recorded
+  /// after which the video record will immediately finish and the [File] will be returned
   static Future<File> pickVideo({
     @required ImageSource source,
+    double maxDuration
```

Comment:
Can you switch the type to Duration, which was basically made for this purpose: https://master-docs-flutter-io.firebaseapp.com/flutter/dart-core/Duration-class.html.

Reply:
I have done this now